images/egress/router/egress-router.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# OpenShift egress router setup script
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+BLANK_LINE_OR_COMMENT_REGEX="([[:space:]]*$|#.*)"
+PORT_REGEX="[[:digit:]]+"
+PROTO_REGEX="(tcp|udp)"
+IP_REGEX="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+"
+
+if [[ "${EGRESS_SOURCE:-}" !~ ^${IP_REGEX}$ ]]; then
+    echo "EGRESS_SOURCE unspecified or invalid"
+    exit 1
+fi
+if [[ -z "${EGRESS_DESTINATION:-}" ]]; then
+    echo "EGRESS_DESTINATION unspecified"
+    exit 1
+fi
+if [[ "${EGRESS_GATEWAY:-}" !~ ^${IP_REGEX}$ ]]; then
+    echo "EGRESS_GATEWAY unspecified or invalid"
+    exit 1
+fi
+
+set -x
```

Comment:
Yes and no; it was there before, just earlier in the script. But I guess it's only useful for debugging purposes; if an actual error occurs then it should log something that will show up in "oc logs" anyway. So I'll make it not do that by default.